app/components/search_bar/search_bar.android.js

```diff
@@ -3,176 +3,225 @@
         placeholder: 'Search',
         showCancelButton: true,
-        placeholderTextColor: '#bdbdbd',
-        textColor: '#212121',
+        placeholderTextColor: changeOpacity('#000', 0.5),
         onSearchButtonPress: () => true,
         onCancelButtonPress: () => true,
         onChangeText: () => true,
-        onFocus: () => true,
-        onBlur: () => true
+        onFocus: () => true
     };
 
     constructor(props) {
         super(props);
         this.state = {
-            isOnFocus: false,
-            value: this.props.text
+            isFocused: false,
+            value: ''
         };
-        this.onFocus = this.onFocus.bind(this);
-        this.onBlur = this.onBlur.bind(this);
-        this.onCancelButtonPress = this.onCancelButtonPress.bind(this);
-        this.onSearchButtonPress = this.onSearchButtonPress.bind(this);
-        this.onChangeText = this.onChangeText.bind(this);
     }
 
-    onSearchButtonPress() {
-        const onSearchButtonPress = this.props.onSearchButtonPress;
-        if (this.state.value) {
-            onSearchButtonPress(this.state.value);
+    cancel = () => {
+        this.onCancelButtonPress();
+    };
+
+    onSearchButtonPress = () => {
+        const {onSearchButtonPress} = this.props;
+        const {value} = this.state;
+
+        if (value && onSearchButtonPress) {
+            onSearchButtonPress(value);
         }
-    }
+    };
 
-    onCancelButtonPress() {
-        const onCancelButtonPress = this.props.onCancelButtonPress;
-        this.setState({
-            isOnFocus: false,
-            value: ''
-        });
+    onCancelButtonPress = () => {
+        const {onCancelButtonPress} = this.props;
 
-        onCancelButtonPress();
         Keyboard.dismiss();
-    }
+        InteractionManager.runAfterInteractions(() => {
+            this.setState({
+                isFocused: false,
+                value: ''
+            }, () => {
+                if (onCancelButtonPress) {
+                    onCancelButtonPress();
+                }
+            });
+        });
+    };
 
-    onChangeText(value) {
-        const onChangeText = this.props.onChangeText;
+    onChangeText = (value) => {
+        const {onChangeText} = this.props;
         this.setState({value});
-        onChangeText(value);
-    }
+        if (onChangeText) {
+            onChangeText(value);
+        }
+    };
 
-    onFocus() {
-        const onFocus = this.props.onFocus;
-        this.setState({isOnFocus: true});
-        onFocus();
-    }
+    onFocus = () => {
+        const {onFocus} = this.props;
 
-    onBlur() {
-        const onBlur = this.props.onBlur;
-        this.setState({isOnFocus: false});
-        onBlur();
-        Keyboard.dismiss();
-    }
+        this.setState({isFocused: true});
 
-    blur = () => {
-        this.onBlur();
+        if (onFocus) {
+            onFocus();
+        }
     };
 
     render() {
         const {
-            height,
+            autoCapitalize,
```

Comment:
We might be able to consider having this component pass through all unrecognized props to the TextInput (like having `const {myProp1, myProp2, ...otherProps} = this.props` and then having `<TextInput {...otherProps} .../>` as the actual component) or by having the TextInput with its props be passed in from the parent and then add any extra props we need after the fact instead of having all of them explicitly defined by this component